### ISSUE TITLE
Improve on usage of `inj{SortX{}, SortKItem{}}` as a way to produce symbols

### ISSUE THREAD

**Issue by Baltoli** [runtimeverification/llvm-backend#904]
Currently, our printing infrastructure relies on injecting non-block terms (ints, bools, etc.) into `KItem` in order to have them work with the generated-code recursion scheme in the LLVM backend. For example, the proof hint code serialises the term `2::Int` as:
```
inj{SortInt{}, SortKItem{}}(\dv{SortInt{}}("2"))
```

This causes a problem when considering the implementation of #905; we will be deserializing these terms from the trace and passing them as arguments to functions that _don't expect the outer injection_. We therefore need to strip the outer constructor when deserializing the binary to patterns.

However, we can't unconditionally strip injections into `KItem` from the outside; it may be that a function relies on such an injection to match:
```
syntax Foo ::= foo()
syntax KItem ::= f(KItem) [function]
rule f(A) => A
```
Here, `f(foo())` _needs_ the inner term to be `inj{SortFoo{}, SortKItem{}}(...)`.

There are two possible solutions here:
1. Introduce a symbol to `domains.md` that signals to serialization code that it should _always_ be stripped when producing a pattern in this context. (easy)
2. Adapt the recursion scheme code to deal better with non-symbol terms. (hard)

I propose implementing (1.) specifically for the proof trace context; that way terms will end up being:
```
#rawTerm(\dv{SortInt{}}("2"))
```
and the binding can transparently strip the outer constructor.

**Comment by Baltoli** [comment]
I've done some experimenting with this, and the solution is actually a combination of the two existing strategies. We need to:
1. Add a symbol `KItem ::= #serialized(KItem)` to `BASIC-K` (so that it's always in scope).
2. Instead of emitting `inj{SortX{}, SortKItem{}}{...}` for raw terms, emit `#serialized{}(inj{...}(...))`.
3. In the _client parser code_, look for `#serialized` terms that will end up being function arguments, then strip two layers off the pattern to get the actual un-injected argument.

**Comment by Baltoli** [comment]
Closed by: https://github.com/runtimeverification/llvm-backend/pull/907

There is some outstanding work to do re. migrating existing usages of the spurious injection pattern to the new `rawTerm` version, but the actual feature has been implemented.

**Comment by Robertorosmaninho** [comment]
> Closed by: #907
> 
> There is some outstanding work to do re. migrating existing usages of the spurious injection pattern to the new `rawTerm` version, but the actual feature has been implemented.

Do we have an issue to track the outstanding work left to do?

**Comment by Baltoli** [comment]
https://github.com/runtimeverification/llvm-backend/issues/916
